src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -0,0 +1,180 @@
+        {
+            resultEx = null;
+            readyForResponse = new SemaphoreSlim(0);
+            _correlationId = AndroidBrokerHelper.GetValueFromBrokerPayload(brokerPayload, BrokerParameter.CorrelationId);
+            //Need to disable warning for non awaited async call.
+#pragma warning disable 4014
+            try
+            {
+                Task.Run(() => AcquireTokenInternalAsync(brokerPayload));
+            }
+            catch
+            {
+                _logger.Error("Broker Operation Failed to complete.");
+                throw;
+            }
+#pragma warning restore 4014
+            await readyForResponse.WaitAsync().ConfigureAwait(false);
+            return resultEx;
+        }
+
+        private async Task AcquireTokenInternalAsync(IDictionary<string, string> brokerPayload)
+        {
+            _brokerHelper.SayHelloToBroker(_activity);
+
+            Context mContext = Application.Context;
+
+            brokerPayload[BrokerParameter.BrokerAccountName] = AndroidBrokerHelper.GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LoginHint);
+            _logger.InfoPii(
+                "It switched to broker for context: " + mContext.PackageName + " login hint: " +
+                AndroidBrokerHelper.GetValueFromBrokerPayload(brokerPayload, BrokerParameter.BrokerAccountName),
+                "It switched to broker for context");
+
+            // Don't send silent background request if account information is not provided
+            bool hasAccountNameOrUserId = !string.IsNullOrEmpty(AndroidBrokerHelper.GetValueFromBrokerPayload(brokerPayload, BrokerParameter.BrokerAccountName)) ||
+                !string.IsNullOrEmpty(AndroidBrokerHelper.GetValueFromBrokerPayload(brokerPayload, BrokerParameter.Username));
+            if (hasAccountNameOrUserId)
+            {
+                _logger.Verbose("User is specified for silent token request");
+                string silentResult = _brokerHelper.GetAuthTokenSilently(brokerPayload, _activity);
+                resultEx = CreateMsalTokenResponseFromResult(silentResult);
+                readyForResponse.Release();
+                return;
+            }
+            else
+            {
+                _logger.Verbose("User is not specified for silent token request");
+            }
+
+            _logger.Verbose("Starting Authentication Activity");
+
+            if (resultEx == null)
+            {
+                _logger.Verbose("Initial request to broker");
+                // Log the initial request but not force a prompt
+            }
+
+            // onActivityResult will receive the response for this activity.
+            // Lauching this activity will switch to the broker app.
+            Intent brokerIntent = _brokerHelper.GetIntentForInteractiveBrokerRequest(brokerPayload, _activity);
+            if (brokerIntent != null)
+            {
+                try
+                {
+                    _logger.Verbose(
```

Comment:
is any of this pii? can these be info logs instead?

Reply:
no pii here. just app info. i can update